File: apps/api/src/CodeStackLMS.Domain/Entities/Lesson.cs
using CodeStackLMS.Domain.Common;
using CodeStackLMS.Domain.Enums;

namespace CodeStackLMS.Domain.Entities;

public class Lesson : BaseEntity, IAuditableEntity
{
    public Guid ModuleId { get; set; }
    public string Title { get; set; } = string.Empty;
    public int Order { get; set; }
    public LessonType Type { get; set; }
    
    public string? VideoUrl { get; set; }
    public string? VideoBlobPath { get; set; }
    public VideoSourceType VideoSource { get; set; } = VideoSourceType.None;
    public string? VideoMimeType { get; set; }
    public int? DurationSeconds { get; set; }
    
    public string? TextContent { get; set; }
    
    public string? LinkUrl { get; set; }
    public string? LinkDescription { get; set; }
    
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Module Module { get; set; } = null!;
    public ICollection<LessonArtifact> Artifacts { get; set; } = new List<LessonArtifact>();
}

File: apps/web/src/types/lesson.ts
export interface LessonArtifact {
  id: string;
  fileName: string;
  contentType: string;
  sizeBytes: number;
  downloadUrl: string;
}

export interface LessonResponse {
  id: string;
  moduleId: string;
  title: string;
  order: number;
  lessonType: string;
  videoUrl: string | null;
  createdAt: string;
  artifacts: LessonArtifact[];
}

export interface CreateLessonRequest {
  moduleId: string;
  title: string;
  order: number;
  videoUrl?: string;
}

export interface UpdateLessonRequest {
  title: string;
  videoUrl?: string | null;
}
